pkg/tracer/common/model.go

```diff
@@ -86,8 +90,61 @@ type ConnectionStats struct {
 	Direction        Direction `json:"direction"`
 	State            State     `json:"state"`
 	NetworkNamespace string    `json:"network_namespace"`
-	SendBytes        uint64    `json:"send_bytes"`
-	RecvBytes        uint64    `json:"recv_bytes"`
+
+	SendBytes uint64 `json:"send_bytes"`
+	RecvBytes uint64 `json:"recv_bytes"`
+
+	HttpMetrics []HttpMetric `json:"http_metrics"`
```

Comment:
Good point. I reviewed it before seeing that and didn't realize it at that time. Funny thing is this is also only done like this in this tcp tracer datamodel.

Reply:
Making invariants is quite tricky in Go. In this case we then would need to make json encoding/decoding for the supertype `Metric` because it's being sent over network.
So I liked the option 3 more except the blob part. I see no advantages in doing that, because the process-agent will need to unblob that instead of just traversing the structure.

So maybe like that?
```
type HttpProtocolData struct {
   Metrics ...
}

type ConnectionStats struct {
   HttpProtocolData *HttpProtocolData
}
```
Then if it's defined then it is the http protocol. How about that?

Reply:
True, but the first option is at least an extensible API without needing to change it if we introduce different kinds of metrics/applicationprotocols

Reply:
The suggested change is still tying the stats to the http protocol. This makes the API not extensible

Reply:
The question here is what do we want to aim for:
* Being strict (not being able to send http metrics for non-http connections)
* Being flexible (not having to create a new data type for every supported protocol)

From what I see we can't do both and you're both arguing for the other requirement.